### PR TITLE
ret-304-if-etag-is-if-none-match return 304 if If-None-Match == ETag

### PR DESCRIPTION
**Pull request by yaitskov**
**Please refer to the contribution guidelines in the README when submitting PRs.**

### CODE REVIEW

File: localstack/services/s3/s3_listener.py [hunk 1]
```diff
@@ -510,6 +510,23 @@ def convert_to_chunked_encoding(method, path, response):
...
+        s = s[1:-1]
+    if s[0] == "'" and s[-1] == "'":
+        s = s[1:-1]
+    return s
+
+
+def ret304_on_etag(data, headers, response):
+    etag = response.headers.get('ETag')
+    if etag:
+        match = headers.get('If-None-Match') or headers.get('if-none-match')
```
[the remark this answers is not in the document]
Comment (yaitskov) [reply localstack/services/s3/s3_listener.py:524]: World is not perfect. AWS eats low case version too. Wireshark prooves that.

File: localstack/services/s3/s3_listener.py [hunk 2]
```diff
@@ -510,6 +510,23 @@ def convert_to_chunked_encoding(method, path, response):
     response.headers.pop('Content-Length', None)
 
 
+def unquote(s):
```
[the remark this answers is not in the document]
Comment (yaitskov) [reply localstack/services/s3/s3_listener.py:513]: fixed

File: localstack/services/s3/s3_listener.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (yaitskov) [reply localstack/services/s3/s3_listener.py]: Oh. I see it python http server normalizes HTTP headers nonetheless of original casing.
Fixed.

**Comment by whummer** [comment]
Thanks for updating the PR @yaitskov - LGTM 👍

**Pull request merged**